sample/Pizzeria.Store.Application/Orders/OrderingDomainEventHandler.cs

```diff
@@ -0,0 +1,102 @@
+using System.Text.Json;
+using FreeMediator;
+using Lewee.Application.Mediation.Notifications;
+using Lewee.Domain;
+using Microsoft.Extensions.Logging;
+using Pizzeria.Store.Contracts;
+using Pizzeria.Store.Domain;
+
+namespace Pizzeria.Store.Application.Orders;
+
+public class OrderingDomainEventHandler : INotificationHandler<OrderStartedEvent>
+{
+    private readonly IRepository<Order> orderRepository;
+    private readonly IQueryProjectionService queryProjectionService;
+    private readonly IMediator mediator;
+    private readonly ILogger<OrderingDomainEventHandler> logger;
+
+    public OrderingDomainEventHandler(
+        IRepository<Order> orderRepository,
+        IQueryProjectionService queryProjectionService,
+        IMediator mediator,
+        ILogger<OrderingDomainEventHandler> logger)
+    {
+        this.orderRepository = orderRepository;
+        this.queryProjectionService = queryProjectionService;
+        this.mediator = mediator;
+        this.logger = logger;
+    }
+
+    public async Task Handle(OrderStartedEvent notification, CancellationToken cancellationToken)
+    {
+        this.logger.LogInformation(
+            "Handling OrderStartedEvent for order {OrderId}",
+            notification.OrderId);
+
+        // Get the order with its pizzas to create the query projection
+        var spec = new GetOrderQuerySpec(notification.OrderId);
+        var order = await this.orderRepository.QueryOneAsync(spec, cancellationToken);
+
+        if (order is null)
+        {
+            this.logger.LogWarning(
+                "Order {OrderId} not found when handling OrderStartedEvent",
+                notification.OrderId);
```

Comment:
This should be an error.

If an order started event is being handled, but there's no order in the DB, then there's something very wrong.

Reply:
I think added a separate comment saying that this should just return...for now